src/main/java/io/reactivex/Flowable.java

```diff
@@ -8877,14 +8877,14 @@ public final Disposable forEachWhile(final Predicate<? super T> onNext, final Co
      *  <dd>{@code last} does not operate by default on a particular {@link Scheduler}.</dd>
      * </dl>
      *
-     * @return a Flowable that emits the last item from the source Publisher or notifies Subscribers of an
+     * @return a Single that emits the last item from the source Publisher or notifies Subscribers of an
      *         error
      * @see <a href="http://reactivex.io/documentation/operators/last.html">ReactiveX operators documentation: Last</a>
      */
     @BackpressureSupport(BackpressureKind.UNBOUNDED_IN)
     @SchedulerSupport(SchedulerSupport.NONE)
-    public final Flowable<T> last() {
-        return takeLast(1).single();
+    public final Single<T> last() {
```

Comment:
For this operator it is according to the 1.x docs:

https://github.com/ReactiveX/RxJava/blob/1.x/src/main/java/rx/Observable.java#L7398

> Returns an Observable that emits the last item emitted by the source Observable or notifies observers of
> a `{@code NoSuchElementException}` if the source Observable is empty.

2.x matches the 1.x behavior.